target_iceberg/sinks.py

```diff
@@ -51,34 +53,48 @@ def max_size(self) -> int:
         """
         return self.config.get("max_batch_size", 10000)
 
-    def validation(self) -> None:
-        """Extra fields and Partition Cols validation."""
-        assert bool(self.extra_values) == bool(
-            self.extra_values_types
-        ), "extra_fields and extra_fields_types must be both set or both unset"
-        if self.extra_values:
-            assert (
-                self.extra_values.keys() == self.extra_values_types.keys()
-            ), "extra_fields and extra_fields_types must have the same keys"
-
     def process_record(self, record: dict, context: dict) -> None:
-        record_flatten = (
-            flatten_record(
-                record,
-                flattened_schema=self.flatten_schema,
-                max_level=self.flatten_max_level,
-            )
-            | self.extra_values
+        record_flatten = flatten_record(
+            record,
+            flattened_schema=self.flatten_schema,
+            max_level=self.flatten_max_level,
         )
+        for old_name, new_name in self.column_renames.items():
+            record_flatten[new_name] = record_flatten.pop(old_name)
+        record_flatten = {
+            # Convert decimal values to double to avoid type mismatch exceptions
+            k: float(v) if isinstance(v, Decimal)
+            else v.isoformat() if isinstance(v, datetime)
+            else v
+            for k, v in record_flatten.items()
+        }
+        if not self.skip_add_synced_field:
+            record_flatten = record_flatten | { "synced_ms": self.start_time }
         super().process_record(record_flatten, context)
 
+    def get_spark_type(self, col_type):
+        col_type = col_type[0] if isinstance(col_type, list) else col_type
+        return {
+            "string": StringType(),
+            "integer": IntegerType(),
+            "number": DoubleType(),
+            "boolean": BooleanType(),
+            "timestamp": TimestampType(),
```

Comment:
I think timestamp is not the property type, it usually is type string with date-time format (and sometimes it informs a custom format). https://opis.io/json-schema/2.x/formats.html

Reply:
You are correct, however I'm adding `synced_ms` as an extra column and I'm adding it as an actual datetime object, so I added this "timestamp" definition, so that I can use TimestampType in the final schema.